File: Sort_Colors/submission3.py
class Solution(object):
    def sortColors(self, arr):
        
        low = 0
        mid = 0
        high = len(arr)-1

        while mid <= high:
            
            if (arr[mid]==1):
                mid += 1
            elif (arr[mid]==0):
                arr[mid],arr[low] = arr[low],arr[mid]
                mid += 1
                low += 1
            else:
                arr[mid],arr[high] = arr[high],arr[mid]
                high -= 1
        

File: Find_Minimum_in_Rotated_Sorted_Array/submission3.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int n = nums.size(), s = 0, e = n-1, mid;
        while (s<=e){
            mid = s + (e-s)/2;
            if ((nums[mid]<=nums[(mid+1)%n]) && (nums[mid]<=nums[(mid-1+n)%n])){
                return nums[mid];
            } else if (nums[mid] > nums[e]){
                s = mid + 1;
            } else {
                e = mid - 1;
            }
        }

        return -1;
    }
};